Note names

ejkreboot opened the pull request:
Added config options for note names, note name colors, circle fill colors, circle radius. If less than 7 colors are given, the provided colors will be recycled. 

Example:

```
  var fb = fretboard.Fretboard(
    {
        radius: 7.5,
        showNames: true,
        colors: ["#f94d30", "#555958", "#555958", "#555958", "#555958", "#555958", "#555958"],
        fillColors: ["#f94d30", "#555958", "#555958", "#007fd1", "#555958", "#555958", "#555958"],
        nameColors: ['white', 'yellow', 'yellow', 'yellow', 'yellow',  'yellow', 'yellow'],
    })
```
So that we can do things like this:

<img width="477" alt="image" src="https://github.com/user-attachments/assets/b893c3e2-17f6-463d-a03a-4cc27efb88c0">

ejkreboot committed:
note names, circle config options

Added config options for note names, note name colors, circle fill colors, circle radius. Example:

```
      var fb = fretboard.Fretboard(
        {
            where: id,
            radius: 7.5,
            showNames: true,
            nameColor: "black",
            colors: ["#f94d30", "#555958", "#555958", "#555958", "#555958", "#555958", "#555958"],
            fillColors: ["#f94d30", "#555958", "#555958", "#007fd1", "#555958", "#555958", "#555958"]
            nameColors: ['white', 'black', 'black', 'black', 'black', 'black', 'black'],
            frets: 18
        })
```

ejkreboot committed:
Fixed css path

txels commented on src/index.js:
```diff
@@ -268,7 +293,7 @@ export const Fretboard = function (config) {
       container = container
         .append("g")
-        .attr("transform", "scale(-1,1) translate(-" + instance.width + ",0)");
+        .attr("transform", "scale(-1,1) translate(-" + (instance.width-XMARGIN()) + ",0)");
```
Can you explain what was the issue that made you add this margin?

txels reviewed (commented):
Hey many thanks for this, it looks good overall on a quick glance, but I do not understand the reason for the margin adjustments, which seems unrelated to the rest of the PR.

One small ask: could you add your example in the description to the `examples` folder, and ensure it looks OK left-handed as well?

ejkreboot committed:
note names for left hand view

ejkreboot commented:
I added examples, and I added support for left-handed view.

I also am confused by this change:
```
.attr("transform", "scale(-1,1) translate(-" + (instance.width-XMARGIN()) + ",0)")
```
I did not actually make that change. That is the way the code is in the version installed by `npm install fretboards`, but I see that is not the way it is in the main branch. But if I revert that change, things do not Look Right. 

Is it possible the main branch is not quite in sync with the latest release on npm?

txels commented:
> I added examples, and I added support for left-handed view.

You may have forgotten to commit/push the new example (i.e. I don't see any new or extended file in `demos`)

txels commented:
> I also am confused by this change:
> 
> ```
> .attr("transform", "scale(-1,1) translate(-" + (instance.width-XMARGIN()) + ",0)")
> ```
> 
> I did not actually make that change. That is the way the code is in the version installed by `npm install fretboards`, but I see that is not the way it is in the main branch. But if I revert that change, things do not Look Right.
> 
> Is it possible the main branch is not quite in sync with the latest release on npm?

Hmm I will look into that... however just as an initial reaction, if making a PR to a main branch in any repo, I would use that repo's main branch as the starting point, as it may contain unreleased changes you don't want to accidentally undo. This might be the case here.

The release process for this repo is very manual, so bear with me.

txels commented:
So, the margins issue... I've checked both without and with the extra margins... The difference is that with the margin removal in your PR it looks:

![imatge](https://github.com/user-attachments/assets/13c63001-4914-46d9-801e-ab5c20236e29)

Without it:

![imatge](https://github.com/user-attachments/assets/3a138845-7e8a-4ffb-bfe8-8c3dc790205c)

I can't remember TBH the reason to prefer either option... Probably due to usage in fretshare.com (where I use the library) on a first glance I would think that not having the extra margin (as in your PR) is better as it lets the using app decide. Not sure why that is not in main though. I need more time to find out.

ejkreboot committed:
default colors bug and demo file

ejkreboot committed:
default colors bug and demo file

ejkreboot committed:
add .DS_Store to gitignore for mac users

ejkreboot committed:
reverting margins, updated demo

ejkreboot commented:
> Hmm I will look into that... however just as an initial reaction, if making a PR to a main branch in any repo, I would use that repo's main branch as the starting point, as it may contain unreleased changes you don't want to accidentally undo.

Good point. I don't do a lot of PRs, as you can tell by now ;)

I have reverted the margins to match what is in the main repo.

And I did indeed forget to `git add .` The demo file is now in the repo.

txels commented:
Hi, thanks again for this... you made me realise I had some mess in there (a tagged version that was not in main, can't figure out where I released that from). I solved conflicts, cleaned up and tried to set things up to work.

As I hadn't touched this repo in a long time, I had quite some issues with versions and build not working. I realised I don't have a good documentation for reliably setting up a dev environment and TBH I'm not good at JS build tools so there was some fumbling involved. But everything seems to work now (manage to package, release and test at least), I released a 0.6.0 version which should include this.

Let me know if you spot some issue with it.

txels commented:
BTW I ended up re-including the margin adjustments which I did in the past for a reason. I just released the version without pushing to main :facepalm:

txels commented:
Demo is up and works: https://fretboard.txels.com/demos/note_names_colors.html
and you are credited in https://github.com/txels/fretboards/graphs/contributors

ejkreboot commented:
Thanks! (Yeah I also had a little struggle getting the build to work, but got it eventually...the node ecosystem changes fast!)

Latest release looks great.

This is a really nice package. Thank you for developing it.
